microcosm_flask/errors.py

```diff
@@ -4,21 +4,28 @@
 """
 from logging import getLogger
 
-from flask import jsonify
+from marshmallow import fields, Schema
 from werkzeug.exceptions import default_exceptions
 
+from microcosm_flask.conventions.encoding import dump_response_data
+
 
 error_logger = getLogger("errors")
 
 
-def with_headers(error, headers):
-    setattr(error, "headers", headers)
-    return error
+class SubErrorSchema(Schema):
+    message = fields.String(required=True)
+
 
+class ErrorContextSchema(Schema):
+    errors = fields.List(fields.Nested(SubErrorSchema), required=True)
 
-def with_context(error, context):
-    setattr(error, "context", context)
-    return error
+
+class ErrorSchema(Schema):
```

Comment:
I've updated the error schema to match conventions we've discussed offline. Ping @adamhadani